### ISSUE TITLE
panic when using wrong detach keys

### ISSUE THREAD

**Issue by vieux** [docker/cli#380]
`docker run --cpus 0.5 --name test -itd busybox sh -c "ping google.com"`

`docker attach --detach-keys="ctrl-A,a" test`

the attach works on 17.06 but panics on 17.07

```
                                                                                  panic: runtime error: invalid memory address or nil pointer dereference
                                                                                                                                                         [signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x4139376]

                                                     goroutine 12 [running]:
                                                                            bufio.(*Reader).writeBuf(0x0, 0x4fad960, 0xc42037ccc0, 0xc42002fe48, 0x400f69d, 0x49c3b40)
                                                                                                                                                                      	/usr/local/go/src/bufio/bufio.go:504 +0x26
                                      bufio.(*Reader).WriteTo(0x0, 0x4fad960, 0xc42037ccc0, 0x5540198, 0x0, 0x1)
                                                                                                                	/usr/local/go/src/bufio/bufio.go:462 +0x43
                                                                                                                                                                  io.copyBuffer(0x4fad960, 0xc42037ccc0, 0x4fad260, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
                                                                          	/usr/local/go/src/io/io.go:380 +0x31a
                                                                                                                     io.Copy(0x4fad960, 0xc42037ccc0, 0x4fad260, 0x0, 0x0, 0x0, 0x0)
        	/usr/local/go/src/io/io.go:360 +0x68
                                                    github.com/docker/cli/cli/command/container.(*hijackedIOStreamer).beginOutputStream.func1(0xc420405e30, 0xc42045ac20, 0xc42041af60)
           	/go/src/github.com/docker/cli/cli/command/container/hijack.go:123 +0x200
                                                                                        created by github.com/docker/cli/cli/command/container.(*hijackedIOStreamer).beginOutputStream
          	/go/src/github.com/docker/cli/cli/command/container/hijack.go:139 +0x83
```

(sorry for the broken output)

@tiborvass is currently investigating

/cc @jlhawn

**Comment by tiborvass** [comment]
@vieux It's caused by https://github.com/moby/moby/commit/26231b29e7881d25822bffd740d5f73fc2687460

**Comment by aaronlehmann** [comment]
Is this a panic that would have been caught by integration tests if we ran them against the current CLI version instead of fixing the CLI version at an arbitrary point?

**Comment by seemethere** [comment]
This error still exists with Docker 17.07.0-ce

**Comment by vieux** [comment]
@seemethere yes, updated the code on moby/moby, but the fix the issue we need to vendor this code in docker/cli

**Comment by dnephin** [comment]
The fix was vendored as part of of #424

We just need it cherry-picked in docker-ce (https://github.com/docker/docker-ce/pull/221)

**Comment by andrewhsu** [comment]
@vieux should be `--cpus`:
```
docker run --cpus 0.5 --name test -itd busybox sh -c "ping google.com"
```

**Comment by vieux** [comment]
@andrewhsu right, but the panic still exists.

**Comment by vieux** [comment]
fixed in #424
